### PR TITLE
PCHR-3834: Update Global Styles

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of the PR the Global Styles has been updated of the SSP theme according to the new mockup.
Related PRs
https://github.com/compucorp/civihr-employee-portal/pull/502
https://github.com/compucorp/org.civicrm.shoreditch/pull/112

#### Brand Colors
```scss
$brand-primary: #0071bd !default;
$brand-secondary: #9494a5 !default;
$brand-success: #44cb7e !default;
$brand-warning: #e6ab5e !default;
$brand-danger: #cf3458 !default;
```
#### Gray Colors
```scss
$gray-darker: #3e3e54 !default;
$gray-dark: #464354 !default;
$gray: #4d4d69 !default;
$gray-light: #9494a5 !default;
$gray-lighter: #c2cfd8 !default;

$light: #fff !default;
$light-pale: #e8eef0 !default;
$light-gray: #f3f6f7 !default;
```
Other than the bootstrap variables, one more gray color is found in the mockup, it is mentioned as
```scss
$chr-gray-1: #4e4e78;
```
In future if other colors are required, the same format needs to be followed

#### Gradients
For the Save button the following gradient variables are created
```scss
$chr-gradient-1: linear-gradient(315deg, #01A2D5 0%, #0070BB 100%);
$chr-gradient-2: linear-gradient(315deg, #0187B1 0%, #005B97 100%);
```
The above gradients will be used in multiple places, for example 
Borders in https://projects.invisionapp.com/d/main#/console/13248050/297993949/preview
In Payroll Details https://projects.invisionapp.com/d/main#/console/13248050/297230123/preview

So there is no known pattern at this moment. So for simplicity's sake, the names are numbered. Future gradients should be added similarly.

#### Fixing Adhoc Colors
Previously throughout the theme, many adhoc colors were used, all of those are removed and variables from `scss/base/_variables.scss` has been used.

#### Font Size
The new font sizes are mapped to bootstrap variables
```scss
$font-size-base: 13px;
$font-size-small: 13px;
$font-size-large: 18px;

$font-size-h1: 24px;
$font-size-h2: 24px;
$font-size-h3: 24px;
$font-size-h4: $font-size-large;
$font-size-h5: $font-size-large;
$font-size-h6: $font-size-large;
```
And a new variable is created for font-size, which could not be mapped to bootstrap
```scss
$chr-font-size-medium: 14px;
```

#### Font Weights
Variables has been created/updated for Font weights
```scss
$crm-font-weight-light: 300;
$crm-font-weight-regular: 400;
$crm-font-weight-semi-bold: 600;
$crm-font-weight-bold: 700;
```

#### Fixing Adhoc Font
Previously throughout the theme, many adhoc font-sizes and weights were used, all of those are removed and variables from `scss/base/_variables.scss` has been used.

### Radio Button Styles
Radio buttons styles were previously declared for Onboarding Wizard only in `_onboarding-wizard.scss`, those has been moved to `drupal/_form.scss`.

### Buttons
In the new design, all buttons other than inside modals, should follow the outline styles.
outlined-buttons
And the Submit buttons should follow a gradient style, so `btn-gradient` is created
gradient-buttons

```scss
.btn-gradient {
  @include button-size($padding-large-vertical, $padding-large-horizontal, $font-size-base, $line-height-large, $border-radius-large);

  background: $chr-gradient-1;
  border: 0;
  color: $light;

  &:hover {
    background: $chr-gradient-2;
    color: $light;
  }
}
```

1. `hook_button()` has been copied from `themes/radix/includes/form.inc` and added in `civihr_default_theme/includes/form.inc`.
2. 'btn-primary-outline'` class has been added for all buttons which are not inside modal and is not a submit button. Previously `'btn-default'` was applied.
3. `btn-gradient` class has been applied to all buttons which are inside modal and is a submit button. Previously `'btn-primary'` was applied.

According to Jon, new gradient buttons might be added in future, but he is not sure of that now, so the architecture is made keeping that in mind. For example, we can rename `btn-gradient` to `btn-primary-gradient` if we add other brand colored gradients in future.

## How it looks
The following images show how different components look like(PS they are not final as we dont have final mockups, this is show how it looks after changing global colors and font sizes)

all-components
2018-06-26 at 6 22 pm
2018-06-26 at 6 24 pm

### Removed unused variables
```scss
$crm-table-heading-padding:      13px 20px !default;	
$crm-table-heading-bg:           #f7f7f9 !default;	
$crm-table-heading-border-color: #dadcde !default;	
	
$crm-table-nested-indent-color:  #dbdddf !default;	
$crm-table-nested-indent-width:  10px !default;
```

## Comment
As part of this PR a generic global styles has been applied and variables has been created for colors and font related things. We still dont have mockups for all individual elements so some elements might look a but odd or not right. But as the architecture is set here, it will be easier to update individual components when we have mockups.

Also Leave and Absence has not been touched yet, as first of all The Styles are used in Admin also, and secondly the mockups are not available.

The css changes affected a lot of files so `--no-verify` suffix has been used to ignore linting errors.

### CODE REVIEW

**Comment by jamienovick** [comment]
This looks fine as a starting point. Did the panels and background colours get updated at all?

**Comment by deb1990** [comment]
> This looks fine as a starting point. Did the panels and background colours get updated at all?

@jamienovick The background color for the body has changed slightly(according to the Mockup), but the panel are still the same, those are planned to be changed as part of PCHR-3667

**Comment by AkA84** [comment]
@jamienovick the panel's *colours* should be changed as well as part of this PR

File: civihr_default_theme/includes/form.inc [hunk 1]
```diff
@@ -3,3 +3,37 @@
  * @file
  * Theme and preprocess functions for forms
  */
+
+
+/**
+ * Implements theme_button().
+ */
+function civihr_default_theme_button($variables) {
+    $element = $variables['element'];
```
Comment (AkA84) [inline civihr_default_theme/includes/form.inc]: the indentation looks wrong

File: civihr_default_theme/includes/form.inc [hunk 2]
```diff
@@ -3,3 +3,37 @@
  * @file
  * Theme and preprocess functions for forms
  */
+
+
+/**
+ * Implements theme_button().
```
Comment (AkA84) [inline civihr_default_theme/includes/form.inc:9]: document what you're doing

File: civihr_default_theme/includes/form.inc [hunk 3]
```diff
@@ -3,3 +3,37 @@
...
+ */
+function civihr_default_theme_button($variables) {
```
Comment (AkA84) [inline civihr_default_theme/includes/form.inc]: i think i need you to annotate here which bits are from radix and which bits you added yourself, cause i don't want to spend time reviewing code that comes straight from radix

File: civihr_default_theme/scss/base/_custom.scss [hunk 4]
```diff
@@ -3,16 +3,17 @@ legend {
   border-bottom: 1px solid $legend-border-color;
   color: $legend-color;
   display: block;
-  font-size: ($font-size-base * 1.5);
+  font-size: $font-size-base;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_custom.scss:8]: we don't need to do this size manipulation anymore?

File: civihr_default_theme/scss/base/_custom.scss [hunk 5]
```diff
@@ -3,16 +3,17 @@ legend {
...
   line-height: inherit;
   margin-bottom: $line-height-computed;
   padding: 0;
   width: 100%;
 }
 
 label {
+  color: $gray-light;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_custom.scss:16]: what was the color before?

File: civihr_default_theme/scss/base/_custom.scss [hunk 6]
```diff
@@ -258,6 +259,10 @@ body {
   width: $width;
 }
 
+.views-exposed-form label {
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_custom.scss]: shouldn't it be 
```scss
views-exposed-form
  label {
```
isn't there a `.views-exposed-form` already available where you can put the `label` style?

File: civihr_default_theme/scss/base/_mixins.scss [hunk 7]
```diff
@@ -61,13 +61,13 @@
 
 @mixin fa($unicode, $font-size: inherit) {
   &:before {
+
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_mixins.scss]: empty line

File: civihr_default_theme/scss/base/_variables.scss [hunk 8]
```diff
@@ -1,5 +1,9 @@
 @import url('//fonts.googleapis.com/css?family=Lato:400,300,300italic,400italic,700,700italic');
 
+
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: empty line

File: civihr_default_theme/scss/base/_variables.scss [hunk 9]
```diff
@@ -1,5 +1,9 @@
...
+$chr-gradient-1: linear-gradient(315deg, #01A2D5 0%, #0070BB 100%);
+$chr-gradient-2: linear-gradient(315deg, #0187B1 0%, #005B97 100%);
+
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: i believe these gradients should be place right after the colors list, given that they're still colors

File: civihr_default_theme/scss/base/_variables.scss [hunk 10]
```diff
@@ -1,5 +1,9 @@
 @import url('//fonts.googleapis.com/css?family=Lato:400,300,300italic,400italic,700,700italic');
 
+
+$chr-gradient-1: linear-gradient(315deg, #01A2D5 0%, #0070BB 100%);
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: shouldn't the start/end colors of the gradients part of the palette (= be one of the color variables)

File: civihr_default_theme/scss/base/_variables.scss [hunk 11]
```diff
@@ -10,38 +14,29 @@ $crm-header-height: 60px !default;
...
+$gray-darker:         #3e3e54 !default;
+$gray-dark:           #464354 !default;
+$gray:                #4d4d69 !default;
+$gray-light:          #9494a5 !default;
+$gray-lighter:        #c2cfd8 !default;
 
-$text-muted:          rgba($gray, 0.6);
+$light:               #fff !default;
+$light-pale:          #e8eef0 !default;
+$light-gray:          #f3f6f7 !default;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: please double check, but i believe that these `$light-` colors are not part of bootstrap. If that's the case, then they should be prefixed with `$chr-`

File: civihr_default_theme/scss/base/_variables.scss [hunk 12]
```diff
@@ -10,38 +14,29 @@ $crm-header-height: 60px !default;
...
 
-$crm-gray-light-alt:  #f0f1f4 !default; // carousel item bg, hr resource bg
-$crm-gray-light-blue: #757b8e !default; // scrollbar thumb, pivot table text
-$crm-gray-darker-alt: #404454 !default; // page title
-$crm-grizzly:         #dcdddd !default;
-$crm-color-neutral:   #949caa !default; // gray shade
-$crm-grayish-blue:    #f4f7f8 !default;
-$crm-gray-matte:      $crm-color-neutral; // disabled font
+$chr-gray-1:          #4e4e78;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: would be better to use an adjective, rather than a number. lighters? darkest? etc

File: civihr_default_theme/scss/base/_variables.scss (hunk 11 above)
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: also please check that the format that bootstrap follows is `$color-shade`, so it should be `$gray-light`. The problem is that `$gray-light` already exist, so you need to understand what kind of shade of gray we're dealing with here, so you can properly name it

File: civihr_default_theme/scss/base/_variables.scss [hunk 13]
```diff
@@ -50,12 +45,24 @@ $fa-font-path:            '../fonts';
...
+$font-size-large:         18px;
+
+$font-size-h1:            24px;
+$font-size-h2:            24px;
+$font-size-h3:            24px;
+$font-size-h4:            $font-size-large;
+$font-size-h5:            $font-size-large;
+$font-size-h6:            $font-size-large;
+
+$chr-font-size-medium: 14px;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: shouldn't you place it right after the default `$font-size-` variables?

File: civihr_default_theme/scss/base/_variables.scss [hunk 14]
```diff
@@ -50,12 +45,24 @@ $fa-font-path:            '../fonts';
...
+$font-size-base:          13px;
+$font-size-small:         13px;
+$font-size-large:         18px;
+
+$font-size-h1:            24px;
+$font-size-h2:            24px;
+$font-size-h3:            24px;
+$font-size-h4:            $font-size-large;
+$font-size-h5:            $font-size-large;
+$font-size-h6:            $font-size-large;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]:
```scss
$font-size-h1:            24px;
$font-size-h2:            $font-size-h2;
$font-size-h3:            $font-size-h2;
$font-size-h4:            $font-size-large;
$font-size-h5:            $font-size-h4;
$font-size-h6:            $font-size-h4;
```

File: civihr_default_theme/scss/base/_variables.scss [hunk 15]
```diff
@@ -50,12 +45,24 @@ $fa-font-path:            '../fonts';
...
+
+$chr-font-size-medium: 14px;
 
-$crm-font-weight-light:   300;
-$crm-font-weight-regular: 400;
-$crm-font-weight-bold:    700;
+
+$crm-font-weight-light:     300;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: shouldn't it be `$chr`

File: civihr_default_theme/scss/base/_variables.scss [hunk 16]
```diff
@@ -50,12 +45,24 @@ $fa-font-path:            '../fonts';
...
+$crm-font-weight-regular:   400;
+$crm-font-weight-semi-bold: 600;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: `semibold`

File: civihr_default_theme/scss/base/_variables.scss [hunk 17]
```diff
@@ -183,13 +184,16 @@ $crm-wizard-active-font-color: $crm-wizard-secondary-color;
...
-$crm-uib-datepicker-header-font-size:       14px;
+$crm-uib-datepicker-header-font-size:       $font-size-base;
 $crm-uib-datepicker-header-font-weight:     $crm-font-weight-regular;
-$crm-uib-datepicker-muted-cell-color:       $crm-grizzly;
+$crm-uib-datepicker-muted-cell-color:       $gray-dark;
 $crm-uib-datepicker-today-bg-color:         $brand-primary;
 $crm-uib-datepicker-today-font-color:       $light;
+
+$crm-jquery-datepicker-week-border:         $light-pale;
+$crm-jquery-datepicker-grid-color:          $light;
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: where are these variables coming from?

File: civihr_default_theme/scss/base/_variables.scss [hunk 18]
```diff
@@ -1,5 +1,9 @@
 @import url('//fonts.googleapis.com/css?family=Lato:400,300,300italic,400italic,700,700italic');
 
+
+$chr-gradient-1: linear-gradient(315deg, #01A2D5 0%, #0070BB 100%);
+$chr-gradient-2: linear-gradient(315deg, #0187B1 0%, #005B97 100%);
```
Comment (AkA84) [inline civihr_default_theme/scss/base/_variables.scss]: do not give indexes, give meaningful names

File: civihr_default_theme/scss/components/_button.scss [hunk 19]
```diff
@@ -35,3 +35,16 @@
...
+
+.btn-gradient {
+  @include button-size($padding-large-vertical, $padding-large-horizontal, $font-size-base, $line-height-large, $border-radius-large);
+
+  background: $chr-gradient-1;
+  border: 0;
+  color: $light;
+
+  &:hover {
+    background: $chr-gradient-2;
```
Comment (AkA84) [inline civihr_default_theme/scss/components/_button.scss]: i thought you were going to darken/lighten it? Defining a gradient just for a hover state seems a bit overkill

File: civihr_default_theme/scss/drupal/_form.scss [hunk 20]
```diff
@@ -400,3 +400,51 @@
     width: 50%;
   }
 }
+
+.form-radios {
```
Comment (AkA84) [inline civihr_default_theme/scss/drupal/_form.scss:408]: > Radio buttons styles were previously declared for Onboarding Wizard only in _onboarding-wizard.scss, those has been moved to drupal/_form.scss.

why was this necessary?

**Review by AkA84 — CHANGES_REQUESTED** [review]

File: civihr_default_theme/scss/base/_custom.scss (hunk 4 above)
Comment (deb1990) [reply civihr_default_theme/scss/base/_custom.scss:8]: Yes, as per the plan, we will not be using any custom font sizes anymore. Instead we have fetched all the different font sizes from the mockup, and only those are used.
If Jon suggests a different font size for a specific element, we will update the variables accordingly and use the same.

File: civihr_default_theme/scss/base/_custom.scss (hunk 5 above)
Comment (deb1990) [reply civihr_default_theme/scss/base/_custom.scss:16]: According to new mockup, the labels should be `#9494a5`($gray-light). Previously the label's were using body color(`#757b8e`) when inside the modal's. So defining this globally for all labels.

File: civihr_default_theme/scss/base/_custom.scss (hunk 6 above)
Comment (deb1990) [reply civihr_default_theme/scss/base/_custom.scss]: Yes right, fixed. But there is no other `.views-exposed-form` available. Also moved the code inside `civihr_employee_portal_theme/civihr_default_theme/scss/drupal/_form.scss`.

File: civihr_default_theme/scss/base/_variables.scss (hunk 10 above)
Comment (deb1990) [reply civihr_default_theme/scss/base/_variables.scss]: They are not used anywhere else, so I did not move them to variables. Should I?

File: civihr_default_theme/scss/base/_variables.scss (hunk 18 above)
Comment (deb1990) [reply civihr_default_theme/scss/base/_variables.scss]: Changed to `$chr-primary-gradient` and `$chr-primary-gradient-hover`

File: civihr_default_theme/scss/components/_button.scss (hunk 19 above)
Comment (deb1990) [reply civihr_default_theme/scss/components/_button.scss]: We should keep the variables, otherwise there is a chance of using different darken/lighten percentage values. I have changed the code to define the variables though
```scss
$chr-primary-gradient-stop-1-color:    #01a2d5;
$chr-primary-gradient-stop-2-color:    #0070bb;
$chr-primary-gradient:                 linear-gradient(315deg, $chr-primary-gradient-stop-1-color 0%, $chr-primary-gradient-stop-2-color 100%);
$chr-primary-gradient-hover:           linear-gradient(315deg, darken($chr-primary-gradient-stop-1-color, 7%) 0%, darken($chr-primary-gradient-stop-2-color, 7%) 100%);
```

File: civihr_default_theme/scss/drupal/_form.scss (hunk 20 above)
Comment (deb1990) [reply civihr_default_theme/scss/drupal/_form.scss:408]: Previously when onboarding wizard was done, I did not ask the question that "whether the radio button styles should be made global" and went for the short route i.e. styling only for onboarding wiz, apologies. But now I have confirmed with Jon that the radio button styles are global.

**Comment by deb1990** [comment]
Closing the PR as the ticket is blocked indefinitely. More details can be found on https://compucorp.atlassian.net/browse/PCHR-3676?focusedCommentId=72492&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-72492

Branch should not be deleted
